modules/auth/webauthn/webauthn.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package webauthn
+
+import (
+	"encoding/binary"
+	"encoding/gob"
+	"strings"
+
+	"code.gitea.io/gitea/models/login"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/setting"
+
+	"github.com/duo-labs/webauthn/webauthn"
+)
+
+//WebAuthn represents the global WebAuthn instance
+var WebAuthn *webauthn.WebAuthn
+
+//Init initializes the WebAuthn instance from the config.
+func Init() {
+	gob.Register(&webauthn.SessionData{})
+	WebAuthn = &webauthn.WebAuthn{
+		Config: &webauthn.Config{
+			RPDisplayName: setting.AppName,
+			RPID:          setting.Domain,
+			RPOrigin:      strings.TrimRight(setting.AppURL, "/"),
```

Comment:
Is it OK to trim the last `/`? Maybe the AppURL contains a sub-path, without last `/` it may not work

Reply:
I have to do that because webauthn library trimmed that and will match and validate the URL.

Reply:
Then it's better to have a comment about why it must be trimmed.

Reply:
This value is incorrect for giteas mounted on suburls.